main/src/test/java/org/apache/james/jdkim/DKIMTest.java

```diff
@@ -183,4 +192,168 @@ public InputStream getBodyInputStream() {
             return headers.getBodyInputStream();
         }
     }
+
+    // ======== IGNORE CODE BELOW THIS FOR STARTERS ====== //
+
+    private static class DkimSignatureHeader {
+        private final String name = "DKIM-Signature";
+        private final String value;
+
+        private DkimSignatureHeader(String value) {
+            this.value = value;
+        }
+
+        public String asMimeHeader() {
+            return name + ": " + value;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == null || getClass() != o.getClass()) return false;
+            DkimSignatureHeader that = (DkimSignatureHeader) o;
+            return Objects.equals(name, that.name) && Objects.equals(value, that.value);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(name, value);
+        }
+    }
+
+    /**
+     * source https://datatracker.ietf.org/doc/html/rfc8601#section-2.7.1
+     */
+    private static enum SignatureVerificationResult {
+        // NONE means The message was not signed so it cannot be a signature result.
+        /**
+         * The message was signed, the signature or signatures were acceptable
+         * to the ADMD, and the signature(s) passed verification tests.
+         */
+        PASS,
+        /**
+         * The message was signed and the signature or signatures were
+         * acceptable to the ADMD, but they failed the verification test(s).
+         */
+        FAIL,
+        /**
+         * The message was signed, but some aspect of the signature or
+         * signatures was not acceptable to the ADMD.
```

Comment:
What is an ADMD ?

Reply:
>  ADministrative Management Domain (ADMD)

The descriptions are copy pasted straight from the [
   Message Header Field for Indicating Message Authentication Status RFC] (https://datatracker.ietf.org/doc/html/rfc7001#section-2.6.) 
The full definition for an administrative management domain is given [this section](https://datatracker.ietf.org/doc/html/rfc7001#section-1.2)

A final version of the API should include these references  in the comments of the class